crates/lang/src/events.rs

```diff
@@ -37,3 +39,24 @@ pub trait BaseEvent {
     /// The generated base event enum.
     type Type;
 }
+
+pub struct PrefixedValue<'a, 'b, T> {
+    pub prefix: &'a [u8],
+    pub value: &'b T,
+}
+
+impl<X> scale::Encode for PrefixedValue<'_, '_, X>
+where
+    X: scale::Encode,
+{
+    #[inline(always)]
+    fn size_hint(&self) -> usize {
+        self.prefix.encode().len() + self.value.encode().len()
+    }
+
+    #[inline]
+    fn encode_to<T: scale::Output>(&self, dest: &mut T) {
+        dest.write(self.prefix.encode().as_slice());
+        dest.write(self.value.encode().as_slice());
```

Comment:
using `encode` like this will allocate dynamic heap memory. please  use `encode_to` instead.